dataactbroker/fileRoutes.py

```diff
@@ -162,36 +162,52 @@ def check_current_page(submission):
         submission_id = submission.submission_id
 
         # /v1/reviewData/
+        # Checks that both E and F files are finished
         review_data = sess.query(Job).filter(Job.submission_id == submission_id,
                                              Job.file_type_id.in_([6, 7]), Job.job_status_id == 4)
-        if review_data.count() > 0:
+        if review_data.count() == 2:
             data = {
                 "message": "The current progress of this submission ID is on /v1/reviewData/ page.",
                 "step": "5"
             }
             return JsonResponse.create(StatusCode.OK, data)
 
         # /v1/generateEF/
+        # Checks that all uploads are successful and both D1 and D2 need to have 0 errors so this must == 2 not > 0
         generate_ef = sess.query(Job).filter(Job.submission_id == submission_id, Job.job_type_id == 4,
                                              Job.job_status_id == 4)
-        if generate_ef.count() > 0:
+        # Need to check that cross file is done as well
+        validate_cross_file = sess.query(Job).filter(Job.submission_id == submission_id,
+                                                     Job.file_type_id.in_([4, 5]), Job.job_type_id == 2,
+                                                     Job.number_of_errors == 0, Job.file_size.isnot(None))
+
+        if generate_ef.count() > 0 and validate_cross_file.count() == 2:
             data = {
                 "message": "The current progress of this submission ID is on /v1/generateEF/ page.",
                 "step": "4"
             }
             return JsonResponse.create(StatusCode.OK, data)
 
         # /v1/validateCrossFile/
-        validate_cross_file = sess.query(Job).filter(Job.submission_id == submission_id,
-                                                     Job.file_type_id.in_([4, 5]), Job.job_type_id == 2,
-                                                     Job.number_of_errors == 0, Job.file_size.isnot(None))
-        if validate_cross_file.count() > 0:
+        if validate_cross_file.count() == 2:
             data = {
                 "message": "The current progress of this submission ID is on /v1/validateCrossFile/ page.",
                 "step": "3"
             }
             return JsonResponse.create(StatusCode.OK, data)
 
+        # /v1/generateFiles/
+        generate_files = sess.query(Job).filter(Job.submission_id == submission_id,
+                                                Job.file_type_id.in_([1, 2, 3]), Job.job_type_id == 2,
+                                                Job.number_of_errors == 0, Job.file_size.isnot(None))
```

Comment:
I'm wondering if somewhere here we want to actually check for job_status being finished